### ISSUE TITLE
False positive CallableParameterUseCaseInTypeContextInspection

### ISSUE THREAD

**Issue by viktorprogger** [kalessil/phpinspectionsea#1699]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6 |
| Language level | 8.0 |

Current behaviour
--
With this sample code:
```php
class T
{
}

function test1(T ...$t): void
{
    $t = test2(...$t);
}

/**
 * @return T[]
 */
function test2(T ...$t): array
{
    return $t;
}
```
I have this message on the `$t = test2(...$t);` string:
![image](https://user-images.githubusercontent.com/7670669/109651165-370b3600-7b6f-11eb-8cd3-dbdc69731153.png)

Expected behaviour
--
No error since actual and documented types are identical to received one.

Environment details
---
PhpStorm 2021.1 EAP
Build #PS-211.6222.11, built on March 1, 2021
PhpStorm EAP User
Expiration date: March 31, 2021
Runtime version: 11.0.10+9-b1341.2 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-47-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1980M
Cores: 4
Registry: run.processes.with.pty=TRUE, documentation.show.toolbar=true, debugger.valueTooltipAutoShowOnSelection=true, ide.balloon.shadow.size=0
Non-Bundled Plugins: FrameSwitcher (3.25.1-211), com.alayouni.ansiHighlight (1.2.4), com.chrisrm.idea.MaterialThemeUI (6.1.2), com.intellij.ideolog (203.0.26.0), com.mallowigi (36.0), com.nerdscorner.android.plugin.github (2.0.2), com.tabnine.TabNine (0.3.4), zielu.gittoolbox (203.4.8), Jetbrains TeamCity Plugin (2020.2.85899), org.jetbrains.plugins.node-remote-interpreter (211.6222.2), com.kalessil.phpStorm.phpInspectionsEA (4.0.6), ru.adelf.idea.dotenv (2021.1)
Current Desktop: ubuntu:GNOME

**Comment by ea-inspections-team** [comment]
Thanks for reporting @viktorprogger !
This has been fixed already with changes from #1763.
